src/Test.UnitTests.Sarif/Visitors/ExtractAllArtifactionLocationsVisitorTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Test.Utilities.Sarif;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class ExtractAllArtifactLocationsVisitorTests
+    {
+        [Fact]
+        public void ExtractAllArtifactLocationsVisitor_EmptyLog()
+        {
+            var visitor = new ExtractAllArtifactLocationsVisitor();
+            visitor.VisitRun(new Run());
+            visitor.AllArtifactLocations.Count.Should().Be(0);
+        }
+
+        [Fact]
+        public void ExtractAllArtifactLocationsVisitor_ExtractsMultipleLocations()
+        {
+            var sarifLog = new SarifLog
+            {
+                Runs = new[]
+                {
+                    new Run
```

Comment:
>Run [](start = 24, length = 3)

Add an artifacts table with one item at least. Add a result location that refers to this only by index. Add a second case that has both an index and URI.